src/probe.rs

```diff
@@ -149,41 +149,62 @@ impl<R: Read + Seek> Probe<R> {
 
 	#[allow(clippy::shadow_unrelated)]
 	fn guess_inner(&mut self) -> Result<Option<FileType>> {
```

Comment:
```suggestion
	fn guess_inner(&mut self) -> std::io::Result<Option<FileType>> {
```

We can just return an `io::Result` now